pkg/proxy/iptables/proxier.go

```diff
@@ -898,17 +1008,24 @@ func (proxier *Proxier) syncProxyRules() {
 					"-A", string(fwChain),
 					"-m", "comment", "--comment", fmt.Sprintf(`"%s loadbalancer IP"`, svcName.String()),
 				}
-				// We have to SNAT packets from external IPs.
-				writeLine(natRules, append(args, "-j", string(KubeMarkMasqChain))...)
+
+				// Each source match rule in the FW chain may jump to either the SVC or the XLB chain
+				chosenChain := svcXlbChain
+				// If the service is only proxying to local endpoints, we will not have a double-hop
```

Comment:
This comment would be better as a positive.  If we are proxying globally, we need to masquerade in case we cross nodes.  If we are proxying only locally, we can retain the source IP.